### DISCUSSION TITLE
Running large samples in HSSM (vs HDDM)

### DISCUSSION THREAD

**Discussion by HNP-data** [lnccbrown/HSSM#584] [Q&A]
We have previously used hddm package successfully to run a series of models on a data of 600+ participants with ~ 200 trails each. However, we now have difficulties replicating the same analyses in HSSM on a full sample. 
When we run the same model on a small subsample of the data ( 40 data points total), then the model successfully converges. We iteratively increase sample size to 200 data points total and we started getting an error in the form "There were 3000+ divergencies after tuning. Increase 'target_accept" or reparameterize." 
What are the sample size limits on HSSM. Should we go back to HDDM?
Thank you!

**Comment by AlexanderFengler** [comment]
Hi @HNP-data ,

sorry to hear you are having trouble with HSSM.

Could you provide the model specification you were using with this, so we can help diagnose the problem precisely?

Here are some initial comments:

1. As per your report on divergences, I assume you are using the NUTS sampler and I will assume you are using the basic 'ddm' model with 'analytical' likelihood. We are working actively on robustifying the algorithm we are using for computing 'ddm' likelihoods so that it's gradients (which drive the transitions for the NUTS sampler) are less and less finicky. It is a known problem that the DDM gradients can be unruly for very small 'effective rts' (after deducting non-decision-time). This could be at the core of your problem. However if random subsets of 40 participants work consistently, while all 600 don't I rather suspect that there might be an issue with the actual posterior geometry. As a first pass, I would try both the `noncentered` parameterization as well as the `centered` parameterization (argument `noncentered=True/False`).
2. Actually HSSM allows you to choose the sampler you would like to run from a large array of options. HDDM runs a slice sampler, which you can supply in HSSM models via the `step` argument to `.sample()`. 
3. In HSSM we now also support variational inference, which you might want to try on your problem to see if the results come out reasonable. 

Best,
Alex
